hack/verify-canonical-import-path.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+staging_repos="${KUBE_ROOT}/staging/src/k8s.io"
+
+# verify-canonical-import-path checks that files in the staging directory
+# have doc.go containing canonical import paths.
+# See https://golang.org/doc/go1.4#canonicalimports for more details.
+
+failed=false
+while read i; do
+    # don't consider generated files
+    if grep -q "DO NOT EDIT" $i; then
+        continue
+    fi
+
+    # extract import path and package name from the file name
+    # If i="hack/../staging/src/k8s.io/client-go/scale/doc.go", then
+    # import_path=k8s.io/client-go/scale and package_name=scale
+    import_path=${i:20:-7}
+    package_name=${import_path##*/}
+    
+    if ! grep -q "package ${package_name} // import \"${import_path}\"" $i; then
```

Comment:
Here, we check if the import is correct when the doc.go exists. Do we want to prompt when doc.go is missing too?

Reply:
> Do we want to prompt when doc.go is missing too?

I thought so too, but just looping through subdirectories is not enough since some subdirectories shouldn't have `doc.go`s like `test`, `artifacts`, `Godeps`, `hack`, `testdata`.

Only some directories should be "packages". So `doc.go`s should be subjective and be added only if the directory is intended to be a package.

Reply:
Right